### PR TITLE
Task Submission Card Implementations

### PR DESCRIPTION
**Pull request by simranjeetka**
# Description

This card will be displayed on the dashboard and will display the submission status of assignment, It will let users download their submission work and gives them capability to resubmit the work.

Build and migrate coffee script to Type script  and Enhance API client to support added functions for submission page done by Vishali.
Build UI using angular material components and disable the old module system and integrate with new angular module system  by Simran

Fixes # (issue)

## Type of change

*Please delete options that are not relevant.*

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

*Ran the code on Chrome, Edge and Firefox and monitored console for errors. There were no errors reported.*

## Testing Checklist:

- [x] Tested in latest Chrome
- [x] Tested in latest Edge
- [x] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# Screenshot
screenshot

### CODE REVIEW

**Comment by simranjeetka** [comment]
HI @erickao5858 ,

I have incorporated comments related to unused code and pushed again. I have tried the `--no-verify` option too but I think it has not worked. 

I think we should get rid of post commit hook if the linting is not required. 

Thank you

File: src/app/doubtfire-angularjs.module.ts [hunk 1]
```diff
@@ -89,7 +89,6 @@ import 'build/src/app/projects/states/dashboard/directives/student-task-list/stu
 import 'build/src/app/projects/states/dashboard/directives/directives.js';
 import 'build/src/app/projects/states/dashboard/directives/task-dashboard/directives/task-assessment-card/task-assessment-card.js';
 import 'build/src/app/projects/states/dashboard/directives/task-dashboard/directives/task-outcomes-card/task-outcomes-card.js';
-import 'build/src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.js';
```
Comment (macite) [inline src/app/doubtfire-angularjs.module.ts:92]: This file should also be removed now this is migrated.

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  selector: 'task-submission-card',
+  templateUrl: 'task-submission-card.component.html',
+  styleUrls: ['task-submission-card.component.scss'],
+})
+export class TaskSubmissionnCardComponent implements OnInit {
+  @Input() task: any;
+
+  submission_date: String;
+  public canReuploadEvidence: any;
+  public canRegeneratePdf: any;
```
Comment (macite) [inline src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.ts:14]: Are these really "any"? Why not boolean?

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      style="border: 1px solid black; background-color: #337ab7; color: white"
+      [hidden]="!(canReuploadEvidence || canRegeneratePdf)"
+      mat-button
+      [matMenuTriggerFor]="downloads"
+    >
+      Actions <span class="caret"></span>
+    </button>
+    <!-- #docregion sub-menu -->
+    <mat-menu #downloads="matMenu">
+      <button mat-menu-item (click)="uploadAlternateFiles()">Upload Alternate Files</button>
```
Comment (macite) [inline src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html:42]: Only if they can reupload

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      <button mat-menu-item (click)="regeneratePdf()">Regenerate PDF</button>
```
Comment (macite) [inline src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html:43]: Only if they can regenerate Pdf

**Review by macite — CHANGES_REQUESTED** [review]
Looks like a few changed files are missing.

File: src/app/doubtfire-angularjs.module.ts (hunk 1 above)
Comment (vishali1111) [reply src/app/doubtfire-angularjs.module.ts:92]: The build folder is ignored and not checked in, but i have removed the files.

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html (hunk 3 above)
Comment (vishali1111) [reply src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html]: Done

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html (hunk 4 above)
Comment (vishali1111) [reply src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.html]: Done

File: src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.ts (hunk 2 above)
Comment (simranjeetka) [reply src/app/projects/states/dashboard/directives/task-dashboard/directives/task-submission-card/task-submission-card.component.ts]: These re code changes done.

**Pull request closed without merging**